Makefile

```diff
@@ -624,9 +624,17 @@ $(OBJDIR)/ceed.pc : pkgconfig-prefix = $(prefix)
 	    -e "s:%prefix%:$(pkgconfig-prefix):" \
 	    -e "s:%libs_private%:$(pkgconfig-libs-private):" $< > $@
 
+ifeq ($(filter install,$(MAKECMDGOALS)),install)
+  CFLAGS += $(if $(ASAN),$(AFLAGS)) -DCEED_JIT_SOURCE_ROOT="\"$(abspath $(DESTDIR)$(includedir))/\""
+else
+  CFLAGS += -DCEED_JIT_SOURCE_ROOT="\"$(abspath $(includedir))/\""
+endif
+
```

Comment:
The makefile magic is the last piece needed for this to work. I'm experimenting and working on this, but makefiles are one of my weak areas

Reply:
I think the missing piece is recompilation.

When I run
```
prefix=/home/jeremy/Dev/test HIP_DIR=/opt/rocm make prove -j search=t411
```
the Hip backend can find and load the QFunction source correctly
```
...
JiT Source Root: 
/home/jeremy/Dev/test/include/

----- Loading QFunction User Source -----

---------- Ceed JiT ----------

Current source file: 
/home/jeremy/Dev/test/include/ceed-jit-source/gallery/ceed-identity.h
```

and Ratel was able to successfully build against this version of libCEED

Reply:
It may be easiest for user if the JIT path is defined in a libceed config header that is installed with libceed. During building this file can use the source JIT path, but when installed it will need to be updated with the installed JIT path.

Reply:
I'm terrible at this sort of makefile/install manipulation, so if you have a fast commit that can do it, please feel free to add it to my branch - I'm breaking for lunch for the next hour